Update for Stardew Valley 1.4, internal refactoring

Pathoschild opened the pull request:
This pull request updates the code for SMAPI 3.0 (still compatible with current versions of SMAPI). Let me know if you want me to change anything!


This pull request...

* Updates the code for SMAPI 3.0 and Stardew Valley 1.4.
* Updates the carpenter menu for the new options in Stardew Valley 1.3.
* Updates the mod build package.
* Migrates to the new simplified `.csproj` format.

Let me know if you want me to change anything!

Pathoschild committed:
update for SMAPI 3.0

SlyAceZeta commented:
@Pathoschild Your commit is still not fully 3.0 compliant. I'm getting a "legacy events" error.

`[SMAPI] Rented Tools will break in the upcoming SMAPI 3.0 (legacy events are deprecated since SMAPI 2.9).`

Pathoschild commented:
@Azekthi Can you [upload your SMAPI log](https://log.smapi.io) (see instructions on that page) and post a link here?

SlyAceZeta commented:
Sure, here you go: https://log.smapi.io/GXYgizJb

SlyAceZeta commented:
Whoops, please disregard my comment. It completely skipped my mind that I couldn't just copy/paste your updated files, I have to actually compile them to make them work. 😅

Pathoschild committed:
update mod build package & migrate to new .csproj format

The new package version adds support for SMAPI 3.0 and Stardew Valley 1.4. This commit also sets the target platform to x86, to avoid mismatched processor architecture warnings since the game is 32-bit.

Pathoschild committed:
update for Stardew Valley 1.4

Pathoschild committed:
update for carpenter changes in Stardew Valley 1.3

Pathoschild committed:
bump versions
